tools/pdf-export/default.nix

```diff
@@ -0,0 +1,16 @@
+{buildNpmPackage, chromium, python3}:
+buildNpmPackage (let
+  version = "v0.1.0";
+in {
+  inherit version;
+  src = ./.;
+  nativeBuildInputs = [python3];
+  pname = "html-pdf-export";
+  PUPPETEER_SKIP_CHROMIUM_DOWNLOAD = 1;
+  npmDepsHash = "sha256-DC9f++84BPNX6UoTSK8ZPJNDBIPbodbmBqoKyPr9ZiE=";
+  dontNpmBuild = true;
+  postInstall = ''
```

Comment:
post install? is this due to nix not packaging node very cleanly? Does this break the seal? whats going on here?

Reply:
Each derivation is built in a few steps.  These steps allow for modification of other derviation builds, so in this case the npm derivation does work to prep (moves src into the working directory), patch (nothing here since it's not defined in npm build, as far as I know), prebuild (copies the node_modules from a derviation that handles fetching the dependencies), build (does nothing because we disable it with the no build flag), install (copies the entire project into `$out/lib` and the thing defined in `bin` in the `package.json` into `$out/bin`) and then postinstall (the thing defined here)  so this allows for doing a slight mod to our output bin that injects the chrome binary location as an environment variable